pyxem/signals/segments.py

```diff
@@ -0,0 +1,444 @@
+        else:
+            segment_intensities = np.sum(self.segments.data, axis=(1, 2))
+            for i in range(len(correlated_vectors)):
+                correlated_vector_intensities[i] = np.zeros(
+                    len(correlated_vector_indices[i]))
+                for n, index in zip(range(len(correlated_vector_indices[i])),
+                                    correlated_vector_indices[i]):
+                    correlated_vector_intensities[i][n] = np.sum(
+                        segment_intensities[index])
+
+        vdfseg = VDFSegment(Signal2D(correlated_segments),
+                            DiffractionVectors(correlated_vectors),
+                            correlated_vector_intensities)
+
+        # Transfer axes properties of segments
+        vdfseg.segments = transfer_signal_axes(vdfseg.segments, self.segments)
+        n = vdfseg.segments.axes_manager.navigation_axes[0]
+        n.name = 'n'
+        n.units = 'number'
+
+        return vdfseg
+
+    def get_virtual_electron_diffraction(self, calibration, shape, sigma=None):
+        """ Obtain a virtual electron diffraction signal that consists
+        of one virtual diffraction pattern for each segment. The virtual
+        diffraction pattern is composed of Gaussians centered at each
+        vector position. If given, the integrated intensities of each
+        vector will be taken into account by multiplication with the
+        Gaussians.
+
+        Parameters
+        ----------
+        calibration : float
+            Reciprocal space calibration in inverse Angstrom per pixel.
+        shape : tuple
+            Shape of the signal, (shape_x, shape_y) in pixels, where
+            shape_x and shape_y are integers.
+        sigma : float
+            The standard deviation of the Gaussians in inverse Angstrom
+            per pixel.
+
+        Returns
+        -------
+        virtual_ed : ElectronDiffraction2D
+            Virtual electron diffraction signal consisting of one
+            virtual diffraction pattern for each segment.
+        """
+        vectors = self.vectors_of_segments.data
+        segments = self.segments.data
+        num_segments = np.shape(segments)[0]
+
+        if self.intensities is None:
+            print("The VDFSegment does not have the attribute intensities."
+                  "All intensities will be set to ones.")
+            intensities = np.ones_like(vectors)
+        else:
+            intensities = self.intensities
+
+        if sigma is None:
+            sigma = calibration
+
+        size_x, size_y = shape[0], shape[1]
+        cx, cy = -size_x / 2 * calibration, -size_y / 2 * calibration
+        x, y = np.indices((size_x, size_y))
+        x, y = x * calibration + cx, y * calibration + cy
+        virtual_ed = np.zeros((size_x, size_y, num_segments))
+
+        for i in range(num_segments):
+            if np.shape(np.shape(vectors[i]))[0] <= 1:
+                virtual_ed[..., i] = get_gaussian2d(
```

Comment:
This condition needs a test.

Reply:
I deleted it, it's an odd edge case, and refactoring to sim as signal is a better answer. (https://github.com/dnjohnstone/pyxem/pull/4)